File: python/LeetCode/1-two-sum.py
def twoSum(arr,target):

    for i in range(len(arr)):
        for j in range(len(arr)):
            if i==j:
                break

            if arr[i]+arr[j] == target:
                return [i,j]





arr =  [2,7,11,15]
target = 9
print(twoSum(arr,target))

File: javascript/Leet-Code/1-two-sum.js
twoSum = function (arr, target) {
  let map = {};

  for (let i in arr) {
    let temp = target - arr[i];
    if (map[temp]) {
      return [parseInt(map[temp]), parseInt(i)];
    }

    map[arr[i]] = i;
  }
};

// Two Sum with Hash Map
// We can also solve this by sorting and using Two Pointers
